### PR TITLE
Added Responses With Wrong HTTP Status Code query for OpenAPI #2897

### LINKED ISSUE #2897: Add  Responses With Wrong HTTP Status Code query for OpenAPI
### Platform
OpenAPI 3.0

### Description
HTTP Responses status code should be in range of [200-599]

### PR DESCRIPTION
**Pull request by joaoReigota1**
Signed-off-by: João Reigota <joao.reigota@checkmarx.com>

Closes #2897

**Proposed Changes**
- Add Responses With Wrong HTTP Status Code query for OpenAPI 

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: assets/queries/openAPI/responses_wrong_http_status_code/query.rego [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package Cx
+
+import data.generic.openapi as openapi_lib
+
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) != "undefined"
+	response := doc.paths[n][oper].responses[code]
```
Comment (felipe-avelar) [inline assets/queries/openAPI/responses_wrong_http_status_code/query.rego:8]: I think we should check responses on [component object](https://swagger.io/specification/#responses-object)...

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Check comments, please

File: assets/queries/openAPI/responses_wrong_http_status_code/query.rego (hunk 1 above)
Comment (joaoReigota1) [reply assets/queries/openAPI/responses_wrong_http_status_code/query.rego:8]: I think responses in component object don't use http status code, they are used for ref, ex:
```
"responses": {
    "NotFound": {
      "description": "Entity not found."
    },
    "IllegalInput": {
      "description": "Illegal input for operation."
    },
    "GeneralError": {
      "description": "General Error",
      "content": {
        "application/json": {
          "schema": {
            "$ref": "#/components/schemas/GeneralError"
          }
        }
      }
    }
...
"get": {
    "responses": {
         "400": {
             "$ref": "#components/responses/NotFound"
         }
     }
}
```

Comment (felipe-avelar) [reply assets/queries/openAPI/responses_wrong_http_status_code/query.rego:8]: Make sense

**Review by felipe-avelar — APPROVED** [review]
LGTM

**Pull request merged**